JavaScript/JavaScriptSDK/TelemetryContext.ts

```diff
@@ -145,12 +176,35 @@ module Microsoft.ApplicationInsights {
 
             sessionStateEnvelope.time = Util.toISOStringForIE8(new Date(timestamp));
 
-            tc._track(sessionStateEnvelope); 
+            tc._track(sessionStateEnvelope);
+        }
+
+        private _applyCustomProperties(envelope: Microsoft.Telemetry.Envelope) {
+            if (this.properties && envelope && envelope.data && (<any>envelope.data).baseData) {
+                var telemetryItem = (<any>envelope.data).baseData;
+                telemetryItem.properties = telemetryItem.properties || {};
+                Object.getOwnPropertyNames(this.properties).forEach((val, index, array) => {
+                    if (telemetryItem.properties[val] === undefined) {
+                        telemetryItem.properties[val] = this.properties[val];
+                    }
+                });
+            }
+        }
+
+        private _applyCustomMeasurements(envelope: Microsoft.Telemetry.Envelope) {
+            if (this.measurements && envelope && envelope.data && (<any>envelope.data).baseData) {
```

Comment:
Agree, but when I start making it one function it's becoming hard/ugly to make it abstract from concrete member - properties or measurements. So I like it the way it's now better (still open for concrete suggestions/code examples)